docs/credentials.adoc

```diff
@@ -1,16 +1,130 @@
-= cerdentials.zip file format
+= credentials.zip file format
 
-To authenticate on the ostree server, aktualizr uses cerdentials.zip archive.
+The credentials.zip file provides the link between a user's on-line account on ATS Garage or OTA+ deployment and developer's machine. It is used for:
 
-The minimal requirement for this archive is including `treehub.json` file with
-the following content:
+* Uploading the built OSTree image to treehub with garage-push
+* Offline-signing Uptane images targets metadata with garage-sign
+* Provisioning the device with Aktualizr or aktualizr_cert_provider
+* Providing the URL to the update server to Aktualizr (this is templated into sota.toml at build time, to allow software updates to modify that URL)
 
+The following files are present in credentials.zip:
+
+[options="header"]
+|======================
+| Filename in zip | Purpose | Used by
+| treehub.json | Location and authentications for treehub and Uptane repo | garage-sign, garage-push
+| client.crt | Certificate for TLS client authentication | Aktualizr 
+| client.key | Private key for TLS client authentication | Aktualizr
+| root.crt | Root CA for TLS client authentication | Aktualizr
+| repo.url
+|======================
+
+== treehub.json
+
+The treehub.json file points to the treehub installation, and also includes authentication information that is used for both treehub and Uptane metadata uploads.
+
+The general format is like this:
+
+```
+{
+  "ostree": {
+    "server": "https://localhost:2443/api/v3"
+  },
+  // Authentication information
+}
+```
+
+Multiple types of authentication are supported, and more types may be added in the future.  Clients should ignore unrecognised authentication types, and the server may add multiple authentication types to support both old and new clients.
+
+=== No authentication (e.g. community edition)
+
+```
+{
+  "ostree": {
+    "server": "https://localhost:2443/api/v3"
+  },
+  "no_auth" : {},
```

Comment:
Why is this an empty object I didn't get this in the end ?

Reply:
I was thinking that if we need to add any options, then we can put something in that object

Reply:
Would be happy with anything else though.

Reply:
I think this should be either true or false? I don't think there is anything that could be added that would make no_auth be less confusing ?

Reply:
`true` works for me